test/src/cpp-integration-query-condition.cc

```diff
@@ -994,6 +994,75 @@ TEST_CASE(
   query.finalize();
   array.close();
 
+  if (vfs.is_dir(array_name))
+    vfs.remove_dir(array_name);
+}
+
+TEST_CASE(
+    "Testing dense query condition with overlapping fragment domains",
+    "[query][query-condition][dense][overlapping-fd]") {
+  Context ctx;
+  VFS vfs(ctx);
+
+  if (vfs.is_dir(array_name))
+    vfs.remove_dir(array_name);
+
+  // Create a simple 1D vector with domain 1-10 and one attribute.
+  Domain domain(ctx);
+  domain.add_dimension(Dimension::create<int>(ctx, "d", {{1, 10}}, 5));
+  ArraySchema schema(ctx, TILEDB_DENSE);
+  schema.set_domain(domain).set_order({{TILEDB_ROW_MAJOR, TILEDB_ROW_MAJOR}});
+  Attribute attr = Attribute::create<int>(ctx, "a");
+
```

Comment:
NIT: Why does the new line separate after the attribute is created? Seems a little unintuitive to me, as opposed to putting the new line before attr is created.